sys/Makefile.include

```diff
@@ -111,3 +111,20 @@ endif
 ifneq (,$(filter clif, $(USEMODULE)))
   INCLUDES += -I$(RIOTBASE)/sys/clif/include
 endif
+
+# zptr can use pointer compression of four byte aligned pointers, by right-
+# shifting them by two and stuffing in an uint16_t.
+# That only works on 32bit platforms, if total RAM size is <=2**18 and the
+# start address is known.
+ifneq (,$(filter zptr,$(USEMODULE)))
+  ifeq (,$(filter arch_8bit arch_16bit,$(FEATURES_USED)))
```

Comment:
This looks like a very complex way of something that should be a `FEATURES_REQUIRED += arch_32bit`~~, and what should be in a `makefile.dep`~~
I assume this is to make this fully optional/transparent to whether the CPU supports it or not.

Reply:
> `FEATURES_REQUIRED += arch_32bit`

Well, that would make applications using "zptr" compile only on 32bit archs.
It should just transparently compile out (not compress pointers if not possible).
But this can be changed to `ifneq (,$(filter arch_32bit,$(FEATURES_USED)))`.

Reply:
> Well, that would make applications using "zptr" compile only on 32bit archs.

Yeah, I realized that 5 secs later